pyproject.toml

```diff
@@ -49,7 +49,7 @@ test = [
   "pytest-aiohttp==1.0.5",
   "pytest-cov==5.0.0",
   "ruff==0.3.4",
-  "safety==3.0.1",
+  "safety==3.1.0",
```

Comment:
Where are we using safety?

Reply:
Hm, yeah it seems unused. Let's remove it instead.

Reply:
Not entirely sure (we should check with @frenck) but isnt this used for some of the pre commit rules ?
https://github.com/home-assistant-libs/python-matter-server/blob/main/.pre-commit-config.yaml#L69

Reply:
Isn't the detect-private-key just one of the builtin hooks?

https://github.com/pre-commit/pre-commit-hooks

Reply:
Hmm, in that case I have no idea but we should double check because this came with the set of filter rules/config Frenck used as base

Reply:
At a quick glance, it doesn't look like this hook imports safety: https://github.com/pre-commit/pre-commit-hooks/blob/main/pre_commit_hooks/detect_private_key.py :thinking:

Reply:
Let's remove it then and see what breaks!

Reply:
PR #661